cpp/src/decisiontree/decisiontree_impl.cuh

```diff
@@ -75,6 +77,47 @@ void print_node(const std::string &prefix,
   }
 }
 
+template <typename T>
+std::string to_string_high_precision(T x) {
+  static_assert(std::is_floating_point<T>::value || std::is_integral<T>::value,
+                "T must be float, double, or integer");
+  std::ostringstream oss;
+  oss.imbue(std::locale::classic());  // use C locale
+  if (std::is_floating_point<T>::value) {
+    oss << std::setprecision(std::numeric_limits<T>::max_digits10) << x;
+  } else {
+    oss << x;
+  }
+  return oss.str();
+}
+
+template <class T, class L>
+std::string dump_node_as_json(
+  const std::string &prefix,
+  const std::vector<SparseTreeNode<T, L>> &sparsetree, int idx) {
+  const SparseTreeNode<T, L> &node = sparsetree[idx];
+
+  std::ostringstream oss;
+  if ((node.colid != -1)) {
+    oss << prefix << "{\"nodeid\": " << idx
+        << ", \"split_feature\": " << node.colid
+        << ", \"split_threshold\": " << to_string_high_precision(node.quesval)
+        << ", \"yes\": " << node.left_child_id
```

Comment:
Does the yes/no syntax match some existing tree dumping format? Or maybe avoiding true and false since they sometimes get converted to bools by json libs?

Reply:
XGBoost's `dump_model()` uses this notation.